ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventReaderQueries.scala

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.common
+
+import anorm.{RowParser, SqlParser}
+import com.daml.lf.data.Ref.Party
+import com.daml.lf.ledger.EventId
+import com.daml.lf.value.Value.ContractId
+import com.daml.platform.store.backend.EventStorageBackend
+import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
+import com.daml.platform.store.backend.common.SimpleSqlAsVectorOf._
+import com.daml.platform.store.dao.events.Raw
+import com.daml.platform.store.dao.events.Raw.FlatEvent
+import com.daml.platform.store.interning.StringInterning
+
+import java.sql.Connection
+import scala.annotation.tailrec
+
+class EventReaderQueries(
+    queryStrategy: QueryStrategy,
+    stringInterning: StringInterning,
+) {
+  import EventStorageBackendTemplate._
+
+  type EventSequentialId = Long
+
+  case class SelectTable(tableName: String, selectColumns: String)
+
+  def fetchContractIdEvents(contractId: ContractId, requestingParties: Set[Party])(
+      connection: Connection
+  ): Vector[EventStorageBackend.Entry[Raw.FlatEvent]] = {
+
+    val witnessesColumn = "flat_event_witnesses"
+
+    val tables = List(
+      SelectTable(
+        tableName = "participant_events_create",
+        selectColumns =
+          s"$selectColumnsForFlatTransactionsCreate, ${queryStrategy.constBooleanSelect(false)} as exercise_consuming",
+      ),
+      SelectTable(
+        tableName = "participant_events_consuming_exercise",
+        selectColumns =
+          s"$selectColumnsForFlatTransactionsExercise, ${queryStrategy.constBooleanSelect(true)} as exercise_consuming",
+      ),
+    )
+
+    def selectFrom(tableName: String, selectColumns: String) =
+      cSQL"""
+      (
+        SELECT
+          #$selectColumns,
+          event_witnesses,
+          command_id
+        FROM
+        (
+            SELECT
+              #$selectColumns,
+              #$witnessesColumn as event_witnesses,
+              command_id
+            FROM
+              #$tableName
+            WHERE
+              contract_id = ${contractId.coid}
```

Comment:
pls limit with the ledger end, we do not want to expose information which is ahead of the ledger end